src/components/tiles/table/use-grid-context.ts

```diff
@@ -17,9 +17,16 @@ export const useGridContext = ({ content, modelId, showRowLabels, triggerColumnC
   const inputRowId = useRef(uniqueId());
   const dataSet = content.dataSet;
 
-  // this tracks ReactDataGrid's notion of the selected cell
-  const selectedCell = useRef<TPosition>({ rowIdx: -1, idx: -1 });
```

Comment:
Maybe keeping track of this will solve some of the weird behavior related to adding values to the input row of the table. Thanks for pointing this out, I'll give it more thought.

Reply:
I did find a use for the `selectedCell` ref. However, it's a pretty hacky partial solution to a problem, and I wonder if it would be possible to come up with a better solution.

The problem occurs when you press tab to end entering a value in the input row. This causes the RDG selected cell to change, which triggers `onSelectedCellChange`. Even though this happens after the new row is added to the `dataSet`, the `rows` property that gets passed to `useDataSet` does not include the newly added row, so my code gets confused about what's actually selected. So the real solution to this problem is to make sure `rows` is properly updated before `onSelectedCellChange` gets called.

The hacky solution is to call a `setTimeout` after adding the new row which updates RDG's selected cell to be where it was supposed to be after pressing tab. By the time `setTimeout` triggers, `rows` has updated, so it's possible to get the correct location for the selected cell.

Can you think of a way to force the correct `rows` into `onSelectedCellChange`? I guess using a ref for `rows` rather than using a constant?